src/flownet/utils/plot_results.py

```diff
@@ -0,0 +1,372 @@
+
+    Args:
+        vector: Name of the vector to plot.
+        prior_data: List of prior ensemble data DataFrames.
+        posterior_data: List of posterior ensemble data DataFrames.
+        reference_simulation: EclSum object for the reference simulation.
+        plot_settings: Settings dictionary for the plots.
+
+    """
+    plt.figure()  # (figsize=[16, 8])
+
+    if len(prior_data):
+        plot_ensembles("prior", vector, prior_data, plot_settings)
+
+    if len(posterior_data):
+        plot_ensembles("posterior", vector, posterior_data, plot_settings)
+
+    if reference_simulation:
+        plt.plot(
+            reference_simulation.dates,
+            reference_simulation.numpy_vector(vector),
+            color=plot_settings["reference_simulation_color"],
+            alpha=1,
+        )
+
+    if plot_settings["vertical_lines"] is not None:
+        for vertical_line_date in plot_settings["vertical_lines"]:
+            plt.axvline(x=vertical_line_date, color="k", linestyle="--")
+
+    if plot_settings["errors"] is not None:
+        if vector in plot_settings["errors"]:
+            plt.errorbar(
+                plot_settings["errors"][vector][0],
+                plot_settings["errors"][vector][1],
+                yerr=plot_settings["errors"][vector][2],
+                fmt="o",
+                color="k",
+                ecolor="k",
+                capsize=5,
+                elinewidth=2,
+            )
+
+    plt.ylim([plot_settings["ymin"], plot_settings["ymax"]])
+    plt.xlabel("date")
+    plt.ylabel(vector + " [" + plot_settings["units"] + "]")
+    plt.savefig(re.sub(r"[^\w\-_\. ]", "_", vector), dpi=300)
+
+
+def remove_duplicates(df: pd.DataFrame) -> pd.DataFrame:
+    """Remove duplicates for the combination or DATE and REAL.
+
+    Args:
+        df: Input pandas DataFrame with columns: [DATE, REAL, VECTOR1, VECTOR2, ..., VECTOR_N]
+
+    Returns:
+        A cleaned dataframe
+
+    """
+    return df[~df[["DATE", "REAL"]].apply(frozenset, axis=1).duplicated()]
+
+
+def check_args(args):
+    """Helper function to verify input arguments.
+
+    Returns:
+        Nothing
+
+    Raises:
+        ValueError in case the input arguments are inconsistent.
+
+    """
+    if not (len(args.ymin) == 1 or len(args.ymin) == len(args.vectors)):
+        raise ValueError(
+            f"You should either supply a single minimum y-value or as many as you have vectors ({len(args.vectors)}."
+        )
+
+    if not (len(args.ymax) == 1 or len(args.ymax) == len(args.vectors)):
+        raise ValueError(
+            f"You should either supply a single maximum y-value or as many as you have vectors ({len(args.vectors)}."
+        )
+
+    if not (len(args.units) == 1 or len(args.units) == len(args.vectors)):
+        raise ValueError(
+            f"You should either supply a single units label or as many as you have vectors ({len(args.vectors)}."
+        )
+
+    if (
```

Comment:
If prior or posterior is not defined you will run into trouble here, I think? len(None) is not defined. Better with `if not args.prior and not args.posterior and not args.reference_simulation`?

Reply:
Fixed with 

```
    if (
        not args.prior is None
        and args.posterior is None
        and args.reference_simulation is None
    ):
```

And given all those parameters a default of `None` in argparse.